scheduling-manager/src/cucumber/resources/features/scheduling/replace_after_terminate.feature

```diff
@@ -0,0 +1,38 @@
+Feature: Deploying a Daemon environment to a cluster with Replace After Terminate
+
+  Background:
+    Given a Daemon environment named "DaemonEnvironment":
+      | clusterArn        | TestCluster           |
+      | deploymentMethod  | ReplaceAfterTerminate |
+      | taskDefinitionArn | v1                    |
+
+  Scenario: Deploying a new environment for the first time to an empty cluster
+    Given a cluster named "TestCluster" with the following instances and tasks:
+      | instance | tasks |
+      | i-1      |       |
+      | i-2      |       |
+    When the scheduler runs
+    Then it should start the following tasks:
+      | clusterArn  | containerInstanceArn | taskDefinitionArn | group             |
+      | TestCluster | i-1                  | v1                | DaemonEnvironment |
+      | TestCluster | i-2                  | v1                | DaemonEnvironment |
+    And it should not take any further actions
+
+  Scenario: Maintaining running tasks after they're already runnning
+    Given a cluster named "default" with the following instances and tasks:
+      | instance | tasks          |
+      | i-1      | t-1:v1:PENDING |
+      | i-2      | t-2:v1:RUNNING |
+    When the scheduler runs
+    Then it should not take any further actions
+
+  Scenario: Restarting failed tasks
+    Given a cluster named "default" with the following instances and tasks:
+      | instance | tasks          |
+      | i-1      | t-1:v1:STOPPED |
+      | i-2      | t-2:v1:RUNNING |
+    When the scheduler runs
+    Then it should start the following tasks:
+      | containerInstanceArn | taskDefinitionArn | group             |
+      | i-1                  | v1                | DaemonEnvironment |
+    And it should not take any further actions
```

Comment:
should we also test that all the expected tasks are still running?

Reply:
That's implicit in the "should not take any further actions" bit. The scheduling loop is purely functional, it can't modify the state of the cluster apart from the actions it emits.